tests/static/js/formatters-internal/image.js

```diff
@@ -1,106 +1,189 @@
+      let oldFileImgUrl = Formatters.image(oldFileImg, '0x', false).url;
+      expect(oldFileImgUrl).toEqual('https://dyn.mktgcdn.com/f/0/FOO.jpg/fit=contain');
+      let newFileImgUrl = Formatters.image(newFileImg, 'x0', false).url;
+      expect(newFileImgUrl).toEqual('https://dyn.mktgcdn.com/f/FOO.jpg/fit=contain');
+      let euFileImgUrl = Formatters.image(euFileImg, 'x', false).url;
+      expect(euFileImgUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/fit=contain');
+
+      photoImgUrl = Formatters.image(photoImg, 'x', true).url;
+      expect(photoImgUrl).toEqual('https://dyn.mktgcdn.com/p/FOO/fit=contain');
+      photoWithPaddingImgUrl = Formatters.image(photoWithPaddingImg, 'x', true).url;
+      expect(photoWithPaddingImgUrl).toEqual('https://dyn.mktgcdn.com/p/FOO/1.0000/fit=contain');
+      oldFileImgUrl = Formatters.image(oldFileImg, 'x', true).url;
+      expect(oldFileImgUrl).toEqual('https://dyn.mktgcdn.com/f/0/FOO.jpg/fit=contain');
+      newFileImgUrl = Formatters.image(newFileImg, 'x', true).url;
+      expect(newFileImgUrl).toEqual('https://dyn.mktgcdn.com/f/FOO.jpg/fit=contain');
+      euFileImgUrl = Formatters.image(euFileImg, 'x', true).url;
+      expect(euFileImgUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/fit=contain');
     });
   });
 
-  describe('when image is served from EU with no dimensions specified', () => {
-    it('when choosing the biggest image under threshold, use the width and height on the image object if exists', () => {
-      const euImageUrl = Formatters.image(
-        { url: euUrl,
-          width: 1024,
-          height: 768,
-          sourceUrl: 'https://a.mktgcdn.com/p/FOO/516x384.jpg',
-        }, 'x', false).url;
-      expect(euImageUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/width=1024,height=768,fit=contain');
+  describe('when desiredSize is not parseable', () => {
+    it('throws an error', () => {
+      expect(() => {Formatters.image(photoImg, 'ax')}).toThrowError();
     });
 
-    it('when choosing the biggest image under threshold, use dimensions from the sourceUrl if width/height does not exist', () => {
-      const euImageUrl = Formatters.image(
-        { url: euUrl,
-          width: 1024,
-          sourceUrl: 'https://a.mktgcdn.com/p/FOO/516x384.jpg'
-        }, 'x', false).url;
-      expect(euImageUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/width=516,height=384,fit=contain');
+    it('throws an error', () => {
+      expect(() => {Formatters.image(photoImg, 'xa')}).toThrowError();
     });
 
-    it('when choosing the smallest image over threshold, omit width/height if can\'t parse it from the image object', () => {
-      const euImageUrl = Formatters.image({url: euUrl}, 'x', true).url;
-      expect(euImageUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/fit=cover');
+    it('throws an error', () => {
+      expect(() => {Formatters.image(photoImg, '')}).toThrowError();
```

Comment:
can these tests be more specific? it would help if the name explained what error/why it's being thrown. and you should be able to check the error message being thrown with `toThrowError('Invalid desired size')`

Reply:
done